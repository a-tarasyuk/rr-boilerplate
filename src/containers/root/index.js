import React from 'react';
import styles from './styles.css';
import Header from 'components/header';

const Root = ({ children }) => (
  <div className={ styles.root }>
    <Header />
    
    <div className={ styles.content }>
      { children }
    </div>
  </div>
);

Root.propTypes = {
  children: React.PropTypes.node,
};

export default Root;