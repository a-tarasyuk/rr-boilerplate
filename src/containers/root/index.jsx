import { Route } from 'react-router-dom';
import React from 'react';
import Header from '../../components/header';
import NavLink from '../../components/nav-link';
import Home from '../home';
import Features from '../features';
import styles from './styles.css';

const Root = () => (
  <div className={styles.root}>
    <Header>
      <NavLink href={'/'}>Home</NavLink>
      <NavLink href={'/features'}>Libraries/Tools</NavLink>
    </Header>

    <div className={styles.content}>
      <Route path="/" exact component={Home} />
      <Route path="/features" component={Features} />
    </div>
  </div>
);

export default Root;
