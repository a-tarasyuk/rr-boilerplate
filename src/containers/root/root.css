body {
  margin: 0;
  padding: 0;
}

.root {
  width: 800px;
  margin: 0 auto;
}

.content {}
