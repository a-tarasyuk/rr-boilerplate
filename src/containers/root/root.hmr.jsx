import { hot } from 'react-hot-loader';
import Container from './root';

export const Root = hot(module)(Container);
export default Root;
