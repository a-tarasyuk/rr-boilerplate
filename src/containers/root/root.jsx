import { Route } from 'react-router-dom';
import React from 'react';
import { Features } from '../features';
import { NavLink } from '../../components/nav-link';
import { Header } from '../../components/header';
import { Home } from '../home';
import styles from './root.css';

export const Root = () => (
  <div className={styles.root}>
    <Header>
      <NavLink href={'/'}>Home</NavLink>
      <NavLink href={'/features'}>Libraries/Tools</NavLink>
    </Header>

    <div className={styles.content}>
      <Route path={'/'} exact component={Home} />
      <Route path={'/features'} component={Features} />
    </div>
  </div>
);

export default Root;
