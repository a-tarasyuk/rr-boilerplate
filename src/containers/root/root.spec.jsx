import React from 'react';
import { shallow } from 'enzyme';
import { Root } from './root';

describe('<Root />', () => {
  it('matches its snapshot', () => {
    const component = shallow(<Root />);
    expect(component).toMatchSnapshot();
  });
});
