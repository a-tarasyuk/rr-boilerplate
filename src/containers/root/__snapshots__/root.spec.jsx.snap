// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Root /> matches its snapshot 1`] = `
<div
  className="root"
>
  <Header>
    <NavLink
      href="/"
    >
      Home
    </NavLink>
    <NavLink
      href="/features"
    >
      Libraries/Tools
    </NavLink>
  </Header>
  <div
    className="content"
  >
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      path="/features"
    />
  </div>
</div>
`;
